### ISSUE TITLE
Better error message when git branch is missing

### ISSUE THREAD

**Issue by rbren** [FairwindsOps/reckoner#176]
I think that this might be working as intended, but the behavior confused me and led to some pain.

I was using the latest `hackmd` chart in the core helm charts repo, but had a problem with it, so I forked. Thought I'd be able to use my same course.yaml, but setting `repositories` to point to my fork.

Unfortunately it looks like reckoner tries to user the `version` field to check out a branch/tag (which obv wasn't possible when I was hitting a helm repo).

I would suggest one or more of the following improvements:
* fail when the checkout fails, rather than complaining that the chart path wasn't found (the checkout causes the repo to be empty)
* don't overload the `version` field to mean both git branches/tags and Chart.yaml versions
* default to using `master` if the version tag isn't found, but the `Chart.yaml` does contain the requested version.

On reckoner 3.1.0 (latest)

course.yaml
```yaml
namespace: hackmd
repositories:
  rbrenfork:
    git: git@github.com:rbren/charts.git
    path: stable
charts:
  hackmd:
    repository: rbrenfork
    namespace: hackmd
    version: "1.2.4"
```

Command:
```
$ reckoner plot apps/hackmd/course.yaml
2020-02-09 18:28:50 rbren root[16327] INFO Found Helm Version 3.0.2
2020-02-09 18:28:53 rbren root[16327] INFO Installing hackmd
2020-02-09 18:28:53 rbren root[16327] WARNING $HELM_HOME not set. Using ~/.helm
2020-02-09 18:28:57 rbren root[16327] WARNING Cmd('git') failed due to: exit code(1)
  cmdline: git checkout 1.2.4
  stderr: 'error: pathspec '1.2.4' did not match any file(s) known to git.'
2020-02-09 18:28:57 rbren root[16327] WARNING Branch/tag "1.2.4" does not seem to exist!
2020-02-09 18:28:57 rbren root[16327] WARNING Skipping chart hackmd
2020-02-09 18:28:57 rbren root[16327] ERROR Helm upgrade failed on hackmd
2020-02-09 18:28:58 rbren root[16327] ERROR ERROR: Chart failed to install.
2020-02-09 18:28:58 rbren root[16327] ERROR  - hackmd
2020-02-09 18:28:58 rbren root[16327] ERROR Stopping chart installations due to an error! Some of your charts may not have been installed!
⛵🔥 Encountered errors while running the course ⛵🔥

* * * * *

Chart Name: hackmd
Status: Failed
Error Reason: Command Failed with output below:
STDOUT:
STDERR: Error: path "/home/rbren/.helm/cache/archive/git@github_com:rbren_charts_git/stable" not found

COMMAND: helm upgrade --install hackmd /home/rbren/.helm/cache/archive/git@github_com:rbren_charts_git/stable --namespace hackmd --version 1.2.4 -f /tmp/tmpvjoz1twk.yml
```

**Comment by sudermanjr** [comment]
@rbren the use of version as the ref in a git checkout is well documented and intended behavior - https://github.com/FairwindsOps/reckoner/tree/master/docs#charts

I'm a little confused as to your expected behavior here. If you are pulling the chart from git, then the "version" in the `Chart.yml` means nothing. There is no way to "find" a different version. This is the reason for overloading the version field. The only "version" we have available from a git repo is the git ref. 

I do think we could probably make the error messaging around this a little clearer.

**Comment by sudermanjr** [comment]
>Unfortunately it looks like reckoner tries to user the version field to check out a branch/tag (which obv wasn't possible when I was hitting a helm repo).

I'm also a little confused about this statement, because in your example, you are not using a helm repo, you are using a git repo.

**Comment by rbren** [comment]
Ah - to be clearer, I started with this course:

```
namespace: hackmd
charts:
  hackmd:
    namespace: hackmd
    version: "1.2.4"
```

Which hits the helm-stable repo and gets version `1.2.4`. But there was a problem with that chart, so I forked the git repo and wrote

```
namespace: hackmd
repositories:
  rbrenfork:
    git: git@github.com:rbren/charts.git
    path: stable
charts:
  hackmd:
    repository: rbrenfork
    namespace: hackmd
    version: "1.2.4"
```

It wasn't immediately obvious that by switching the repo mechanism, the meaning of the `version` field changed. I think what I would have expected is for it to pull from master, but throw an error if the `Chart.yaml` version didn't match the version I specified there.

I don't think it's worth changing existing behavior if this is WAI. But a better error message would be super helpful.
